datastore.go

```diff
@@ -11,11 +11,19 @@ import (
 	options "github.com/dgraph-io/badger/options"
 	ds "github.com/ipfs/go-datastore"
 	dsq "github.com/ipfs/go-datastore/query"
-	logger "github.com/ipfs/go-log"
+	logger "github.com/ipfs/go-log/v2"
 	goprocess "github.com/jbenet/goprocess"
 )
 
-var log = logger.Logger("badger")
+type badgerLog struct {
+	logger.ZapEventLogger
+}
+
+func (b *badgerLog) Warningf(format string, args ...interface{}) {
+	b.Warnf(format, args...)
+}
+
+var log = &badgerLog{*logger.Logger("badger")}
```

Comment:
This is annoying... maybe we should just revert the change we made to go-log and add `Warning*` functions for backwards compatibility.

Let's just wrap this where we need it (when constructing the badger logger).

Reply:
But we can do that later.